internal/hook/types/healthchecks.go

```diff
@@ -0,0 +1,80 @@
+package types
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	v1 "github.com/garethgeorge/backrest/gen/go/v1"
+	"github.com/garethgeorge/backrest/internal/hook/hookutil"
+	"github.com/garethgeorge/backrest/internal/orchestrator/tasks"
+	"go.uber.org/zap"
+)
+
+type healthchecksHandler struct{}
+
+func (healthchecksHandler) Name() string {
+	return "healthchecks"
+}
+
+func (healthchecksHandler) Execute(ctx context.Context, cmd *v1.Hook, vars interface{}, runner tasks.TaskRunner, event v1.Hook_Condition) error {
+	payload, err := hookutil.RenderTemplateOrDefault(cmd.GetActionHealthchecks().GetTemplate(), hookutil.DefaultTemplate, vars)
+	if err != nil {
+		return fmt.Errorf("template rendering: %w", err)
+	}
+
+	l := runner.Logger(ctx)
+	l.Sugar().Infof("Sending healthchecks message to %s", cmd.GetActionHealthchecks().GetWebhookUrl())
+	l.Debug("Sending healthchecks message", zap.String("payload", payload))
+
+	PingUrl := cmd.GetActionHealthchecks().GetWebhookUrl()
+
+	// Send a "start" signal to healthchecks.io when the hook is starting.
+	if event == v1.Hook_CONDITION_CHECK_START ||
+		event == v1.Hook_CONDITION_PRUNE_START ||
+		event == v1.Hook_CONDITION_SNAPSHOT_START {
+		PingUrl += "/start"
+	}
+
+	// Send a "fail" signal to healthchecks.io when the hook is failing.
+	if event == v1.Hook_CONDITION_UNKNOWN ||
+		event == v1.Hook_CONDITION_ANY_ERROR ||
+		event == v1.Hook_CONDITION_CHECK_ERROR ||
+		event == v1.Hook_CONDITION_PRUNE_ERROR ||
+		event == v1.Hook_CONDITION_SNAPSHOT_ERROR {
```

Comment:
perhaps my only nit is it may be worth declaring some utilities or constants for this e.g. isStartCondition, isErrorCondition, isSuccessCondition so that we can be consistent about this going forward.

These blocks probably also should have handling for `CONDITION_SNAPSHOT_SUCCESS`, `CONDITION_PRUNE_SUCCESS`, and `CONDITION_CHECK_SUCCESS`

Reply:
Yes, I already thought about it,
I can create an array with the constants and the function or, a function only using switch like this example

<pre>
func isStartCondition(event v1.Hook_Condition) bool {
	switch event {
	case v1.Hook_CONDITION_SNAPSHOT_START,
		v1.Hook_CONDITION_PRUNE_START,
		v1.Hook_CONDITION_CHECK_START:
		return true
	default:
		return false
	}
}
</pre>

What do you see better?

Reply:
I've been looking for a more elegant solution, see what you think.

Reply:
Dictionary approach looks good to me!